src/Umbraco.Core/Services/ContentServiceExtensions.cs

```diff
@@ -93,7 +99,8 @@ public static IEnumerable<string> GetAnchorValuesFromRTEs(this IContentService c
         string rteContent)
     {
         var result = new List<string>();
-        MatchCollection matches = AnchorRegex.Matches(rteContent);
+        var rteUnescaped = Regex.Unescape(rteContent);
+        MatchCollection matches = AnchorRegex.Matches(rteUnescaped);
         foreach (Match match in matches)
         {
             result.Add(match.Value.Split(Constants.CharArrays.DoubleQuote)[1]);
```

Comment:
```suggestion
            result.Add(match.Groups[1].Value);
```
No need to be manipulating strings when we already have the capture group and this change is required for the new regex.